crypto/crypto_aes.c

```diff
@@ -63,30 +63,47 @@ static struct aes_test {
 
 /* Test hardware intrinsics against test vectors. */
 static int
-hwtest(struct aes_test * knowngood)
+hwtest(struct aes_test * knowngood,
+    int(* func)(const uint8_t *, size_t, const uint8_t[16], uint8_t *))
 {
-	void * kexp_hw;
-	uint8_t ctext_hw[16];
+	uint8_t ctext[16];
 
 	/* Sanity-check. */
 	assert((knowngood->len == 16) || (knowngood->len == 32));
 
-	/* Expand the key and encrypt with hardware intrinsics. */
+	/* Expand the key and encrypt with the provided function. */
+	if (func(knowngood->key, knowngood->len, knowngood->ptext, ctext))
+		goto err0;
+
+	/* Do the outputs match? */
```

Comment:
Might as well fix this now that we're comparing an output against a known good value rather than comparing two outputs.